### PR TITLE
utils: align macros 

### PR DESCRIPTION
**Pull request by ldts**

### CODE REVIEW

**Comment by ldts** [comment]
@etienne-lms fyi

File: lib/libutils/ext/include/util.h [hunk 1]
```diff
@@ -89,8 +89,9 @@
 /* x has to be of an unsigned type */
 #define IS_POWER_OF_TWO(x) (((x) != 0) && (((x) & (~(x) + 1)) == (x)))
 
-#define ALIGNMENT_IS_OK(p, type) \
-	(((uintptr_t)(p) & (__alignof__(type) - 1)) == 0)
+#define IS_ALIGNED(x, a)		(((x) & ((a) - 1)) == 0)
+#define IS_ALIGNED_WITH_TYPE(x, type) \
```
Comment (jenswi-linaro) [inline lib/libutils/ext/include/util.h:93]: Now we have two alignment checking macros. I'm wondering if we can make this macro more safe by adding a check that the `type` argument really is a type and not `sizeof(type)` or some other easy mistake.
How about something like:
``` C
#define IS_ALIGNED_WITH_TYPE(x, type) \
        (__extension__({ \
                type __is_aligned_y; \
                IS_ALIGNED((uintptr_t)(x), __alignof__(__is_aligned_y)); \
        }))
```

**Comment by ldts** [comment]
AFICS it is all good. Not sure why IBART fails

File: lib/libutils/ext/include/util.h [hunk 2]
```diff
@@ -89,6 +89,11 @@
 /* x has to be of an unsigned type */
 #define IS_POWER_OF_TWO(x) (((x) != 0) && (((x) & (~(x) + 1)) == (x)))
 
+#ifndef __ASSEMBLER__
+#define __ALIGN_MASK(x,mask)		(((x) + (mask)) & ~(mask))
+#define ALIGN(x,a)			__ALIGN_MASK(x, (typeof(x))(a)-1)
```
[the remark this answers is not in the document]
Comment (ldts) [reply lib/libutils/ext/include/util.h:94]: > few space chars missing
> 
> ```c
> #define __ALIGN_MASK(x, mask)		(((x) + (mask)) & ~(mask))
> #define ALIGN(x, a)			__ALIGN_MASK(x, (typeof(x))(a) - 1)
> ```

@etienne-lms I rely too much on checkpatch...sorry about it

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Pull request merged**